lib/dry/logic/predicates.rb

```diff
@@ -129,6 +129,23 @@ def max_size?(num, input)
           input.size <= num
         end
 
+        def bytesize?(size, input)
+          case size
+          when Integer then size == input.bytesize
```

Comment:
It would be better to use `equal?` here since integers are singletons

Reply:
I can make that change. I would make it  to `size?` too, but that would be a breaking change since `1.equal?(1.0) => false`. (or any of the other Numeric variants)

Is this behavior we want for `bytesize?` but not `size?`

Reply:
I didn't even consider the context, this case is always for ints only so no need to consider the coercion